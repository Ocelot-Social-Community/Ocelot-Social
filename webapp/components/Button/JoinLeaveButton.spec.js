import { shallowMount } from '@vue/test-utils'

import Component from './JoinLeaveButton.vue'

const localVue = global.localVue

describe('JoinLeaveButton.vue', () => {
  let propsData, wrapper, mocks

  beforeEach(() => {
    propsData = {
      group: {
        id: 'g-1',
        name: 'Group 1',
      },
      userId: 'u1',
      isMember: false,
      isNonePendingMember: false,
    }
    mocks = {
      $t: jest.fn((t) => t),
    }
  })

  const Wrapper = () => {
    return shallowMount(Component, { propsData, localVue, mocks })
  }

  describe('shallowMount', () => {
    beforeEach(() => {
      wrapper = Wrapper()
    })

    it('renders', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
