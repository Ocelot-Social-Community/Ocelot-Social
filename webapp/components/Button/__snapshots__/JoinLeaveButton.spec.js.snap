// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinLeaveButton.vue shallowMount renders 1`] = `
<base-button-stub icon="plus" size="regular" type="button" class="join-leave-button has-tooltip" data-original-title="null">
  group.joinLeaveButton.join
</base-button-stub>
`;
